middle_end/flambda2/kinds/flambda_arity.ml

```diff
@@ -14,101 +14,40 @@
 (*                                                                        *)
 (**************************************************************************)
 
-type t = Flambda_kind.t list
-
-type arity = t
-
-let nullary = []
+type t = Flambda_kind.With_subkind.t list
 
 let create t = t
 
-let length t = List.length t
-
 let to_list t = t
 
-include Container_types.Make (struct
-  type nonrec t = t
-
-  let compare t1 t2 = List.compare Flambda_kind.compare t1 t2
-
-  let equal t1 t2 = compare t1 t2 = 0
+let compare_ignoring_subkinds t1 t2 =
+  List.compare Flambda_kind.With_subkind.compare
```

Comment:
I think we should add `Flambda_kind.With_subkind.compare_ignoring_subkind` (and `equal_ignoring_subkind`), and use it here instead of mapping. It would both be more efficient (less allocations), and would add a function that will be useful in many other places.

Reply:
Done